source/server/http/admin.cc

```diff
@@ -311,6 +311,43 @@ Http::Code AdminImpl::handlerStats(const std::string& url, Buffer::Instance& res
   return rc;
 }
 
+Http::Code AdminImpl::handlePrometheusStats(const std::string& url, Buffer::Instance& response) {
+  // As stated above, the /metric endpoint also doesn't support timers locally (only via statsd)
+  Http::Code rc = Http::Code::OK;
+  const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+  statsAsPrometheus(server_.stats().counters(), server_.stats().gauges(), response);
+  return rc;
+}
+
+void AdminImpl::statsAsPrometheus(const std::list<Stats::CounterSharedPtr>& counters,
+                                  const std::list<Stats::GaugeSharedPtr>& gauges,
+                                  Buffer::Instance& response) {
+  std::string comma = ",";
+  for (const Stats::CounterSharedPtr& counter : counters) {
+    std::string tags;
+    std::string sep = "";
+    for (const Stats::Tag tag : counter->tags()) {
+      tags += sep;
+      tags += fmt::format("{}={}", tag.name_, tag.value_);
+      sep = comma;
+    }
+    response.add(fmt::format("# TYPE {0} counter\n", counter->tagExtractedName()));
+    response.add(
+        fmt::format("{0}{{{1}}} {2}\n", counter->tagExtractedName(), tags, counter->value()));
```

Comment:
I think you may need to do some escaping here.  I can't seem to find the docs, but I'm pretty sure that the restrictions on names in the format are:

1) names can contain letters and digits and _ but no other characters
2) names cannot begin with a digit

Someone please correct me if I'm wrong on that.

Reply:
This is the link you have been looking for: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Reply:
Thanks!  That's exactly what I was thinking of (but failed to find).

Reply:
Here is the testing output of the stats: https://gist.github.com/lita/8f145f87d4390e050ea4b36951f359ab

From what I can tell, it seems to follow the metric naming rules for Prometheus.